recaf-assembler/src/main/java/me/coley/recaf/assemble/transformer/AstToClassTransformer.java

```diff
@@ -0,0 +1,41 @@
+package me.coley.recaf.assemble.transformer;
+
+import me.coley.recaf.assemble.BytecodeException;
+import me.coley.recaf.assemble.MethodCompileException;
+import me.coley.recaf.assemble.ast.arch.*;
+import me.coley.recaf.assemble.util.*;
+import org.objectweb.asm.tree.*;
+
+public class AstToClassTransformer {
+
+    private final ClassDefinition definition;
+    private final ClassSupplier classSupplier = ReflectiveClassSupplier.getInstance();
+    private final InheritanceChecker inheritanceChecker = ReflectiveInheritanceChecker.getInstance();
+
+    public AstToClassTransformer(ClassDefinition definition) {
+        this.definition = definition;
+    }
+
+    public ClassNode buildClass() throws MethodCompileException {
```

Comment:
Drops a lot of info

Reply:
ClassNode building is currently not in scope of this pr, it is just a remnant of the changes i tried to make for class building